### ISSUE TITLE
cwctl start fails on keychainless system

### ISSUE THREAD

**Issue by tetchel** [eclipse/codewind#2520]
From https://ci.eclipse.org/codewind/job/Codewind/job/codewind-vscode/job/testsInJenkins/47/console
CLI version `2020-03-19-12:11:47`
```
10:41:41  [INFO: 14:41:38.845 CLIWrapper.js:127]: Running CLI command: cwctl --json start -t latest
10:41:41  
10:41:41  [ERRO: 14:41:38.888 CLIWrapper.js:168]: Error running cwctl --json start -t latest
10:41:41  
10:41:41  [ERRO: 14:41:38.888 CLIWrapper.js:170]: Stdout: Debug: false
10:41:41  ==> created file /home/genie.codewind/.codewind/latest/home/genie.codewind/.codewind/docker-compose.yaml
10:41:41  
10:41:41  [ERRO: 14:41:38.888 CLIWrapper.js:176]: Stderr: 2020/03/19 14:41:38 WRITE_FILE_ERROR[204]: {"error":"sec_keyring","error_description":"dbus: invalid bus address (invalid or unsupported transport)"}.
10:41:41  
10:41:41  [DBUG: 14:41:38.889 LocalCodewindManager.js.setState():144]: Local Codewind state changing from Starting to Error starting
10:41:41  
10:41:41  [ERRO: 14:41:38.890 CLILifecycleWrapper.js:194]: Error starting codewind Error running cwctl start: 2020/03/19 14:41:38 WRITE_FILE_ERROR[204]: {"error":"sec_keyring","error_description":"dbus: invalid bus address (invalid or unsupported transport)"}. 
```

I suspect this is caused by the changes for https://github.com/eclipse/codewind/issues/1306

**Comment by rwalle61** [comment]
for `cwctl start` to use the insecure keychain, we need to do 
```
cwctl --insecure-keyring start
```
or 
```
export INSECURE_KEYRING=true
cwctl start
```

I don't think the Jenkinsfile for the `codewind-installer` repo has been updated yet to have this env var

**Comment by tetchel** [comment]
oh, i didn't realize it had to be explicitly enabled. I guess this is the expected behaviour then

**Comment by rwalle61** [comment]
yep, `cwctl` uses the secure system keyring by default, unless we tell it to use our insecure one

**Comment by jopit** [comment]
This change means Codewind 0.11.0 won't start on many Linux systems out of the box, even if the user doesn't need docker registry credentials (i.e. they're not using appsody projects, or are only using public appsody stacks)

Potential fixes
- Add a preference in the IDEs to enable the insecure registry and document that Linux users should set that preference if they can't start codewind
- Add the preference and have the IDE's detect that startup fails due to this issue and offer to set the preference for the user.
- Cwctl adds a `--no-keyring` parameter which bypasses the use of the keyring in PFE, and the IDEs add a preference for both this and the insecure keyring, and when they detect that startup fails due to this issue and offer to set one of the preference for the user.
- Something else?

**Comment by tetchel** [comment]
I think 1) and 2) are good, but I'd like it if cwctl could return a better error (and the same error on all platforms) when a keyring is not found so we can look for that specific error.

**Comment by jopit** [comment]
For 0.11.0, do option 2 - cwctl to produce a better message that the IDEs can use to know what to tell the user.

For the future:
- Change cwctl to not require the keychain on start
- Find a way that cwctl can get PFE to begin using the credentials from the keyring only when needed.
- Consider storing the credentials in memory (PFE)
- When we try to access the credentials and it fails due to the absence of the keyring, the IDEs prompt to use the insecure keychain
- Would need to restart PFE only if the user had originally used the insecure keyring then switched to using a secure keyring.
- Recommend a keyring for Linux users.

**Comment by tetchel** [comment]
adding svt tag because it's platform dependent and most of us don't use linux

**Comment by jopit** [comment]
Note that cwctl is currently returning invalid JSON when this errors out:
`2020/03/19 14:41:38 WRITE_FILE_ERROR[204]: {"error":"sec_keyring","error_description":"dbus: invalid bus address (invalid or unsupported transport)"}`

**Comment by tetchel** [comment]
looks like it's using the logging library instead of just printing to stdout/err

**Comment by tetchel** [comment]
Can we make sure for 0.11 that `cwctl start` does not fail because there's no keychain?

**Comment by tobespc** [comment]
but what would we want to then happen if there is no keychain ?  For 0.11 we agreed that we would just do option 2 
```
add the preference and have the IDE's detect that startup fails due to this issue and offer to set the preference for the user.
```
and all other enhancements would be for future releases

**Comment by tetchel** [comment]
> but what would we want to then happen if there is no keychain ?

The commands which actually use the keychain (registrsecrets, seckeyring update, sectoken get) would fail when they are used

**Comment by tetchel** [comment]
are we going to improve this?

**Comment by tobespc** [comment]
still needs to be addressed @tetchel but right now its not high our list of enhancements as we are going through a defect fixing stage.  Will look at it later when planning for next sprint

**Comment by malincoln** [comment]
did part of this ship in 0.11.0 and the remainder will ship later or all will go in a later release?

**Comment by tobespc** [comment]
this hasn't been applied at all yet so removing from release

**Comment by wtistang** [comment]
I get similar error with codewind plugin 0.11.0 and VS Code 1.44.2 on a Red Hat VM. When trying to start local codewind server. The error is:

cwctl is available at /home/student/.codewind/0.11.0/cwctl
appsody 0.6.0 is available at /home/student/.codewind/0.11.0/appsody
==> Run cwctl --json --insecure connections list
{"schemaversion":1,"connections":[{"id":"local","label":"Codewind local connection","url":"","auth":"","realm":"","clientid":"","username":""}]}
==> End cwctl --json --insecure connections list
==> Run cwctl --json --insecure status
{"status":"uninstalled"}
==> End cwctl --json --insecure status
==> Run cwctl --json --insecure status
{"status":"uninstalled"}
==> End cwctl --json --insecure status
==> Run cwctl --json --insecure status
{"status":"uninstalled"}
==> End cwctl --json --insecure status
==> Run cwctl --json install -t 0.11.0
<Output hidden>
time="2020-04-30T13:08:58-05:00" level=error msg="Error response from daemon: page not found"
time="2020-04-30T13:09:00-05:00" level=error msg="Error response from daemon: page not found"
==> End cwctl --json install -t 0.11.0
==> Run cwctl --json start -t 0.11.0
Debug: false
2020/04/30 13:09:00 WRITE_FILE_ERROR[204]: {"error":"sec_keyring","error_description":"failed to unlock correct collection '/org/freedesktop/secrets/aliases/default'"}. 
==> End cwctl --json start -t 0.11.0

**Comment by jagraj** [comment]
With the workaround documented here (https://www.eclipse.org/codewind/troubleshooting.html#without-a-keychain-or-keyring-codewind-fails-to-run-certain-commands), I am able to start Codewind 0.11.0 fine on RHEL which does not have keychain config by default. 

![Screen Shot 2020-04-30 at 3 00 25 PM](https://user-images.githubusercontent.com/713849/80753014-91c0a500-8afa-11ea-9d5a-92807745a118.png)

**Comment by tetchel** [comment]
We nixed the preference to set the insecure keychain through the IDEs, so I'm removing the IDE tags
